gobblin-data-management/src/main/java/gobblin/data/management/conversion/hive/provider/HdfsBasedUpdateProvider.java

```diff
@@ -30,7 +35,10 @@
 @AllArgsConstructor
 public class HdfsBasedUpdateProvider implements HiveUnitUpdateProvider {
 
-  private FileSystem fs;
+  private final FileSystem fs;
+
+  // Cache modification times of data location to reduce the number of HDFS calls
+  private static final Cache<Path, Long> PATH_TO_MOD_TIME_CACHE = CacheBuilder.newBuilder().maximumSize(2000).build();
```

Comment:
nit: You can also have `expireAfterAccess(...)` if a data location will only be accessed for a small time range.